specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/1.2-preview/iotcentral.json

```
@@ -0,0 +1,5641 @@
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_ReplaceProperties",
+        "summary": "Replace device properties",
+        "description": "Replace all property values of a device by device ID.",
+        "x-ms-examples": {
+          "Update properties": {
+            "$ref": "./examples/devices_properties_update.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device properties.",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "Devices_UpdateProperties",
+        "summary": "Update device properties via patch",
+        "description": "Update property values of a device by device ID.",
+        "x-ms-examples": {
+          "Update properties": {
+            "$ref": "./examples/devices_properties_update.json"
+          }
+        },
+        "consumes": [
+          "application/json",
+          "application/json-patch+json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/Version"
+          },
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device properties patch.",
+            "schema": {
+              "type": "object"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      }
+    },
+    "/devices/{deviceId}/relationships/": {
```

Path must not end with slash.
```suggestion
    "/devices/{deviceId}/relationships": {
```